libs/execution_context/ebpf_native.c

```diff
@@ -188,7 +206,10 @@ _ebpf_native_clean_up_programs(_In_reads_(count_of_programs) ebpf_native_program
                 programs[i].handle, EBPF_OBJECT_PROGRAM, (ebpf_core_object_t**)&program_object));
             ebpf_assert_success(ebpf_program_register_for_helper_changes(program_object, NULL, NULL));
             ebpf_object_release_reference((ebpf_core_object_t*)program_object);
-            ebpf_assert_success(ebpf_handle_close(programs[i].handle));
+            if (close_handles) {
+                ebpf_assert_success(ebpf_handle_close(programs[i].handle));
+            }
+            programs[i].handle = ebpf_handle_invalid;
```

Comment:
should this be inside the `if` above?  If not, add comment explaining why

Reply:
Fixed.